miden/src/tools/mod.rs

```diff
@@ -70,16 +111,34 @@ pub fn analyze(program: &str, inputs: ProgramInputs) -> Result<ProgramInfo, Prog
 
     let mut total_vm_cycles = 0;
     let mut noop_count = 0;
+    let mut asmop_info_list = Vec::new();
 
     for state in vm_state_iterator {
         let vm_state = state.map_err(ProgramError::ExecutionError)?;
         if matches!(vm_state.op, Some(Operation::Noop)) {
             noop_count += 1;
         }
+        if let Some(asmop_info) = vm_state.asmop {
+            match &mut asmop_info_list
+                .binary_search_by_key(&(asmop_info.get_op()), |(a, _, _): &(String, u8, usize)| &a)
```

Comment:
Is specifying the types here necessary?

Reply:
The compiler complained earlier that it couldn't infer types and needed me to specify types. I'm not sure what changed but it doesn't complain anymore if I remove types. I have removed it.